### PR TITLE
chore: Updated Dockerfile's base image

### LINKED ISSUE #32: DockerFile compliant with rpi and future best practises
## Motivation
If we have recently been able to make progress on pyro-engine features, we have to make sure, every updates is fully rpi compliant.

Hence, as a best practise, it would be nice to merge PRs only after after we can prove that it works on the pi !

What do you think ? 

To date, there is an issue building the dockerfile on rpi (see extract of the error below when try to launch the image using the docker compose from the readme  )

Here is the following error, I will try to perform investigation when I have time, but happy to be helped. 

Thanks a lot !

## 🐛 Bug
```
fatal: [rasp3]: FAILED! => {"changed": true, "cmd": "docker-compose up -d --build", "delta": "0:04:57.131083", "end": "2021-04-18 08:37:35.314231", "msg": "non-zero return code", "rc": 1, "start": "2021-04-18 08:32:38.183148", "stderr": "Building web\nThe command '/bin/sh -c set -eux \t&& apt-get update \t&& apt-get install -y --no-install-recommends \t\tca-certificates \t\tnetbase \t&& rm -rf /var/lib/apt/lists/*' returned a non-zero code: 1\nService 'web' failed to build : Build failed", "stderr_lines": ["Building web", "The command '/bin/sh -c set -eux \t&& apt-get update \t&& apt-get install -y --no-install-recommends \t\tca-certificates \t\tnetbase \t&& rm -rf /var/lib/apt/lists/*' returned a non-zero code: 1", "Service 'web' failed to build : Build failed"], "stdout": "Sending build context to Docker daemon    581MB\r\r\nStep 1/14 : FROM nelnatsol/pyroengine:latest\nlatest: Pulling from nelnatsol/pyroengine\nbc51dd8edc1b: Already exists\ndc4aa7361f66: Already exists\nf7d31f0d4202: Already exists\nedfb78ef674e: Already exists\n9630f24835d2: Already exists\n6372657bdecf: Already exists\n840d64f2d153: Already exists\n6d4def9895b6: Already exists\n8487c400a186: Already exists\n45cd9dc77cfb: Pulling fs layer\n7b28cc33b9c5: Pulling fs layer\nffb7ce2e836d: Pulling fs layer\nfadd4f1b5998: Pulling fs layer\n3e58d33e2319: Pulling fs layer\n6f10b9f1b6ca: Pulling fs layer\n3e58d33e2319: Waiting\n6f10b9f1b6ca: Waiting\nfadd4f1b5998: Waiting\nffb7ce2e836d: Verifying Checksum\nffb7ce2e836d: Download complete\n7b28cc33b9c5: Download complete\n3e58d33e2319: Download complete\nfadd4f1b5998: Verifying Checksum\nfadd4f1b5998: Download complete\n6f10b9f1b6ca: Verifying Checksum\n6f10b9f1b6ca: Download complete\n45cd9dc77cfb: Verifying Checksum\n45cd9dc77cfb: Download complete\n45cd9dc77cfb: Pull complete\n7b28cc33b9c5: Pull complete\nffb7ce2e836d: Pull complete\nfadd4f1b5998: Pull complete\n3e58d33e2319: Pull complete\n6f10b9f1b6ca: Pull complete\nDigest: sha256:bbf4a6630317f4288e2bd408a58bce29412a6c3ab091c2fbd1a134317a0de916\nStatus: Downloaded newer image for nelnatsol/pyroengine:latest\n ---> 3b3f9a6fb2d6\nStep 2/14 : WORKDIR /usr/src/app\n ---> Running in 059112f2ca20\nRemoving intermediate container 059112f2ca20\n ---> 6d8dac8483f3\nStep 3/14 : ENV PYTHONDONTWRITEBYTECODE 1\n ---> Running in 2dcbeb205832\nRemoving intermediate container 2dcbeb205832\n ---> 88c4904c0d16\nStep 4/14 : ENV PYTHONUNBUFFERED 1\n ---> Running in 0a82b4963d45\nRemoving intermediate container 0a82b4963d45\n ---> 0af787da9d0c\nStep 5/14 : ENV PYTHONPATH \"${PYTHONPATH}:/usr/src/app/server/src\"\n ---> Running in 3a6c94b1dd7d\nRemoving intermediate container 3a6c94b1dd7d\n ---> 1072cff0cc73\nStep 6/14 : COPY ./server/src/requirements.txt /usr/src/app/requirements.txt\n ---> 0ac23018cad9\nStep 7/14 : ENV PATH /usr/local/bin:$PATH\n ---> Running in 6c0240a03acd\nRemoving intermediate container 6c0240a03acd\n ---> 3793184944d1\nStep 8/14 : ENV LANG C.UTF-8\n ---> Running in 2aae791b4150\nRemoving intermediate container 2aae791b4150\n ---> b2a3774a0300\nStep 9/14 : RUN set -eux \t&& apt-get update \t&& apt-get install -y --no-install-recommends \t\tca-certificates \t\tnetbase \t&& rm -rf /var/lib/apt/lists/*\n ---> Running in 3d1afc4da9b2\n\u001b[91mstandard_init_linux.go:219: exec user process caused: exec format error\n\u001b[0m", "stdout_lines": ["Sending build context to Docker daemon    581MB", "", "Step 1/14 : FROM nelnatsol/pyroengine:latest", "latest: Pulling from nelnatsol/pyroengine", "bc51dd8edc1b: Already exists", "dc4aa7361f66: Already exists", "f7d31f0d4202: Already exists", "edfb78ef674e: Already exists", "9630f24835d2: Already exists", "6372657bdecf: Already exists", "840d64f2d153: Already exists", "6d4def9895b6: Already exists", "8487c400a186: Already exists", "45cd9dc77cfb: Pulling fs layer", "7b28cc33b9c5: Pulling fs layer", "ffb7ce2e836d: Pulling fs layer", "fadd4f1b5998: Pulling fs layer", "3e58d33e2319: Pulling fs layer", "6f10b9f1b6ca: Pulling fs layer", "3e58d33e2319: Waiting", "6f10b9f1b6ca: Waiting", "fadd4f1b5998: Waiting", "ffb7ce2e836d: Verifying Checksum", "ffb7ce2e836d: Download complete", "7b28cc33b9c5: Download complete", "3e58d33e2319: Download complete", "fadd4f1b5998: Verifying Checksum", "fadd4f1b5998: Download complete", "6f10b9f1b6ca: Verifying Checksum", "6f10b9f1b6ca: Download complete", "45cd9dc77cfb: Verifying Checksum", "45cd9dc77cfb: Download complete", "45cd9dc77cfb: Pull complete", "7b28cc33b9c5: Pull complete", "ffb7ce2e836d: Pull complete", "fadd4f1b5998: Pull complete", "3e58d33e2319: Pull complete", "6f10b9f1b6ca: Pull complete", "Digest: sha256:bbf4a6630317f4288e2bd408a58bce29412a6c3ab091c2fbd1a134317a0de916", "Status: Downloaded newer image for nelnatsol/pyroengine:latest", " ---> 3b3f9a6fb2d6", "Step 2/14 : WORKDIR /usr/src/app", " ---> Running in 059112f2ca20", "Removing intermediate container 059112f2ca20", " ---> 6d8dac8483f3", "Step 3/14 : ENV PYTHONDONTWRITEBYTECODE 1", " ---> Running in 2dcbeb205832", "Removing intermediate container 2dcbeb205832", " ---> 88c4904c0d16", "Step 4/14 : ENV PYTHONUNBUFFERED 1", " ---> Running in 0a82b4963d45", "Removing intermediate container 0a82b4963d45", " ---> 0af787da9d0c", "Step 5/14 : ENV PYTHONPATH \"${PYTHONPATH}:/usr/src/app/server/src\"", " ---> Running in 3a6c94b1dd7d", "Removing intermediate container 3a6c94b1dd7d", " ---> 1072cff0cc73", "Step 6/14 : COPY ./server/src/requirements.txt /usr/src/app/requirements.txt", " ---> 0ac23018cad9", "Step 7/14 : ENV PATH /usr/local/bin:$PATH", " ---> Running in 6c0240a03acd", "Removing intermediate container 6c0240a03acd", " ---> 3793184944d1", "Step 8/14 : ENV LANG C.UTF-8", " ---> Running in 2aae791b4150", "Removing intermediate container 2aae791b4150", " ---> b2a3774a0300", "Step 9/14 : RUN set -eux \t&& apt-get update \t&& apt-get install -y --no-install-recommends \t\tca-certificates \t\tnetbase \t&& rm -rf /var/lib/apt/lists/*", " ---> Running in 3d1afc4da9b2", "\u001b[91mstandard_init_linux.go:219: exec user process caused: exec format error", "\u001b[0m"]}
```

### PR DESCRIPTION
**Pull request by MateoLostanlen**
I just finished building the docker images for armv7, I propose a small PR to add the new image to the docker and close #32 and #14

### CODE REVIEW

**Review by fe51 — APPROVED** [review]
Hi Mateo ! ! Glad to see this PR !

Respecting our new best practise, can you certify that this is working on rpi ?


Also, non blocking for this PR, but should we have an arm Dockerfile AND an Dockerfile compliant for usual computers in order to perform tests (dev and CI tests). What do you think ?

**Comment by MateoLostanlen** [comment]
Yes it works ! On my pi at least. 

I saw there is ways to setup github CI ruinning on your raspberry. I will try to setup that after Ardeche lunch

File: server/src/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM nelnatsol/pyroengine:latest
+FROM pyronear/armv7_pyro-engine:latest
```
Comment (frgfm) [inline server/src/Dockerfile:1]: For posterity, we need to update the tag: on dockerhub, we should have 1 repo per github repo (pyroengine)
The tag should point to the version of it

Then you can add the CPU arch if you want in the tag, but there is no use. Docker automatically retrieves the image that fits your arch. Here is an example:
https://hub.docker.com/r/pytorch/pytorch/tags?page=1&ordering=last_updated
pytorch image have the "linux/amd64" arch
https://hub.docker.com/r/pyronear/armv7_pyro-engine/tags?page=1&ordering=last_updated
this one has "linux/arm"

**Review by frgfm — CHANGES_REQUESTED** [review]
Thanks for the PR! Small detail but it will make the difference in the end and it's easy to change, I added a comment about how we should handle the different images on Dockerhub

File: server/src/Dockerfile (hunk 1 above)
Comment (fe51) [reply server/src/Dockerfile:1]: Okay, interesting !
Hence, if I have understood, just need to have the two archs (armv7 and linux) of pyro-engine image on pyroengine dockerhub ?

Comment (MateoLostanlen) [reply server/src/Dockerfile]: @fe51 Yes but for two different use cases 

@frgfm Ok I get how docker hub works now. I was afraid to delete the other image

**Review by frgfm — APPROVED** [review]
Thanks a lot!

**Pull request merged**